agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -29,6 +29,18 @@ def on_weave_start(topic, data)
       self.start
     end
 
+    # Ensure that all service containers are attached and registered in Weave DNS.
+    #
+    # Only called after the network_adapter is running?
```

Comment:
Turns out this is not actually the case. It might be `already_started?`, but not `running?`, if weave failed/crashed during the `Weave#launch`...